src/shims/env.rs

```diff
@@ -111,4 +112,33 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             Ok(-1)
         }
     }
+
+    fn getcwd(
+        &mut self,
+        buf_op: OpTy<'tcx, Tag>,
+        size_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, Scalar<Tag>> {
+        let this = self.eval_context_mut();
+
+        if this.machine.communicate {
+            let tcx = &{this.tcx.tcx};
+
+            let buf = this.force_ptr(this.read_scalar(buf_op)?.not_undef()?)?;
+            let size = this.read_scalar(size_op)?.to_usize(&*this.tcx)?;
+            // If we cannot get the current directory, we return null
+            if let Ok(cwd) = env::current_dir() {
+                // It is not clear what happens with non-utf8 paths here
+                let mut bytes = cwd.display().to_string().into_bytes();
+                // If the buffer is smaller than the path, we return null
+                if bytes.len() as u64 <= size {
+                    // We need `size` bytes exactly
```

Comment:
My reading of the `getcwd` docs suggest that we don't need to write `size` bytes, but just `bytes.len() + 1` bytes, where the last byte is `\0`

Reply:
oh so we should only write those bytes and not anything else in the buffer. Got it